patch-upstream.sh

```diff
@@ -10,6 +10,9 @@ their Lambdas with Lambda Layers configured to export to the X-Ray backend.
 
 END_DOCUMENTATION
 
+# Run unit tests on ADOT lambdacomponents
+make --directory=adot/collector/lambdacomponents
```

Comment:
should we use an explicit target here instead of relaying on the fact that the first target is the test target?
https://www.gnu.org/software/make/manual/html_node/Goals.html

Reply:
Reading through the documentation, as I understand we need make sure that `make --directory=adot/collector/lambdacomponents` always runs regardless of where it is placed in the make file. Please correct me if my assumption was wrong. 

If that is right, can we do something like

```
.DEFAULT_GOAL := default

default:
     @ make --directory=adot/collector/lambdacomponents
```

Reply:
If the components are living in this repo, why not include the uts as a step in the merge workflow?

if we decide to keep this in the script, please use `set -e`  in the bash script so that it fails on error in any of the execute commands.

Reply:
IMO, we should include unit tests as part of the bash script because every workflow runs the bash script as part of the patch process. So this should cover the testing scope and also catch errors when engineers try to run it locally